### PR TITLE
Panic if JS git hash doesn’t match Rust git hash

### LINKED ISSUE #26: Deploy process for bleeding edge users
Since we have users who make PRs to this repo and then want to use the changes in their own repo, we should have a better deployment story. Ideally we want to be able for users to point to a specific Zaplib git commit and use that (both on the Javascript and on the Rust side).

We also want a good story for people who care more about stable releases, but we can worry about that later (maybe make a ticket for that once we've solved the bleeding edge use case).

### PR DESCRIPTION
**Pull request by janpaul123**
Panic if JS git hash doesn’t match Rust git hash

Closes #26.

This is to prevent accidental mistakes where the two parts of Zaplib are
out of sync.

For local development to not suck, I’ve disabled this check in
development Webpack builds.

I also extracted out some common logic in webpack.config.

### CODE REVIEW

File: zaplib/web/webpack.config.js [hunk 1]
```diff
@@ -48,50 +43,40 @@ const browserConfig = (env, argv) => {
...
 
 const nodeJsConfig = (env, argv) => {
   return {
+    ...common(env, argv),
     target: "node",
     entry: {
       /* eslint-disable camelcase */
       // provides a set of polyfills for running in Node.js,
-      // see zaplib.com/docs/basic_tooling.html#jest for more details
+      // see zaplib.com/docs/existing_webapp.html#jest-integration for more details
```
Comment (pankdm) [inline zaplib/web/webpack.config.js:76]: Is the link supposed to work already? It gives 404 for me

**Comment by pankdm** [comment]
Maybe you could also add the documentation somewhere on how to use the latest versions?- e.g. Rust and npm packages seems to have different syntax.

File: zaplib/web/webpack.config.js (hunk 1 above)
Comment (janpaul123) [reply zaplib/web/webpack.config.js:76]: Yeah we haven't deployed recent docs changes yet; I'll do that soon again

**Comment by janpaul123** [comment]
> Maybe you could also add the documentation somewhere on how to use the latest versions?- e.g. Rust and npm packages seems to have different syntax.

Yes, I added documentation in my last PR.

> Does it mean that for local development you still need to rebuild the rust changes every time you commit locally? I feel that would still make the local development suck. Maybe we should have an option to bypass this check altogether for local dev?

Yeah, I guess it would! Ok let me think about that for a moment.

**Comment by janpaul123** [comment]
> then you need to not forget to always have the second tab open with `cargo watch -- cargo zaplib build --workspace` command always running

That doesn't seem too terrible? I could even make a script that starts both this and the webpack server (and maybe even the docs server)?

**Comment by disambiguator** [comment]
@janpaul123 What's the advantage of having this check in local dev at all? This makes sense to me as a protection for consumers of the library, but it's not something I particularly would find useful

**Comment by disambiguator** [comment]
Sure, could just use node env as well.

**Comment by janpaul123** [comment]
Wanna have another look? :)

**Pull request merged**